account_invoice_crm_tag/models/sale_order.py

```diff
@@ -11,3 +12,15 @@ def _prepare_invoice(self):
         if self.tag_ids:
             vals["crm_tag_ids"] = [(6, 0, self.tag_ids.ids)]
         return vals
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_invoice_line(self, **optional_values):
+        """Transfer CRM tags also to invoice lines, as although being a related stored,
+        it's not correctly saved due to the trick/duality of invoice_line_ids/line_ids.
+        """
+        res = super()._prepare_invoice_line(**optional_values)
+        res["crm_tag_ids"] = [(6, 0, self.order_id.tag_ids.ids)]
```

Comment:
Have you checked if this is still needed?

Reply:
I reviewed this and is for Invoices Analysis Report

Reply:
Have you read the docstring, which states the reason?

Reply:
No, I not read this but i try comment the function ```_prepare_invoice_line``` and view the invoice analysis report to review if crm tags are filled in both cases.

if i comment this function the crm tags are not showed in report

Reply:
Well, you should read the docstrings to know the purpose. And this function is only applied when invoicing the sales order, so do you mean that you have commented this method, invoice a sales order, and then check the invoice analysis report? Please also explain all the steps.

Reply:
ok. Yes this is the steps to follow:

**Firts test**

1. Comment this method
2. Create a new sales order with some crm tags.
3. Confirm the sales order and create the invoice document.
4. Go to the billing module and going to the reports section to open the invoice analysis report.
5. Fumble through CRM tags and check if sales order tags are complete

**Result:** the CRM tags aren't showed into invoice analysis report

**Second test**

1. Uncomment this method
2. Create a new sales order with some crm tags.
3. Confirm the sales order and create the invoice document.
4. Go to the billing module and going to the reports section to open the invoice analysis report.
5. Fumble through CRM tags and check if sales order tags are complete

**Result:** the CRM tags are showed into invoice analysis report